acquia_lift_profiles/acquia_lift_profiles.module

```diff
@@ -459,17 +463,27 @@ function acquia_lift_profiles_node_page_context($node = NULL, $primary = FALSE)
     $account = user_load($node->uid);
     $page_context['author'] = $account->name;
     $page_context['page_type'] = 'node page';
-    // Add the node's taxonomy to the context.
+    // Add the node's vocabulary mappings to the context.
     if (isset($node)) {
-      $results = db_query('SELECT tid FROM {taxonomy_index} WHERE nid = :nid', array(':nid' => $node->nid));
-      $terms = taxonomy_term_load_multiple($results->fetchCol());
-      foreach($terms as $term) {
-        if (isset($flipped[$term->vid])) {
-          $name = $flipped[$term->vid];
-          if (!in_array($term->name, $page_context[$name])) {
-            $page_context[$name][] = $term->name;
-          }
+      foreach ($mappings as $name => $mapping) {
+        if (empty($mapping)) {
+          continue;
+        }
+        list($plugin, $context) = explode(PERSONALIZE_TARGETING_ADMIN_SEPARATOR, $mapping);
+        // Call the module's custom-fetch function to retrieve the terms.
+        $module_name = 'personalize_' . $plugin;
+        $get_terms_function_name = $module_name . '_get_terms';
+        if (!function_exists($get_terms_function_name)) {
+          $error_message = t('You need to enable "@module_name" module and/or define "@function_name" function in the module to custom-fetch the terms into page context.', array (
+            '@module_name' => $module_name,
+            '@function_name' => $get_terms_function_name,
+          ));
+          throw new AcquiaLiftProfilesFunctionNotExistException($error_message);
         }
+        $terms = call_user_func($get_terms_function_name, $node->nid, $context);
+        // Make sure page context terms are unique.
+        $context_terms = array_merge($page_context[$name], $terms);
+        $page_context[$name] = array_unique($context_terms);
       }
```

Comment:
See my comment in the other PR - we shouldn't need to do anything on the PHP side now that these are visitor contexts. In our js we'll grab the context values and set them as the relevant properties on the page context object: https://github.com/acquia/acquia_lift/blob/7.x-1.x/acquia_lift_profiles/js/acquia_lift_profiles.js#L241